Created test data class and http get requests

MariaStrasky opened the pull request:
This branch contains the new class TestDataAccess which implements AbstractDataAccess with some dummy data (right now only a byte array but in the future this will create dummy post objects). It also adds http get requests to the front end javascript so that when the "Top", "Trending" and "Hashtag" buttons are pressed their corresponding controller functions are called.

MariaStrasky committed:
Created test data class and http get requests

MariaStrasky committed:
Merge branch 'master' into Passing-in-test-data-to-front-end

MariaStrasky committed:
save

MariaStrasky committed:
Got http get reading in and displaying byte arrays

MariaStrasky committed:
Revert "save"

This reverts commit 367024e59cabd976fbd2b617f1cd50e41b2bc738.

MariaStrasky committed:
Merging in master to this branch

MariaStrasky committed:
Removing Inject statements to prevent errors

MariaStrasky committed:
Merge branch 'master' into Passing-in-test-data-to-front-end

MariaStrasky committed:
Make my test class match the new abstract class definition

This at least gets rid of some of the compilation errors

reoliveira commented on app/controllers/HashtagContentController.java:
```diff
@@ -15,6 +16,7 @@
  */
 public class HashtagContentController extends Controller {
 
+    private TestDataAccess testDataSource = new TestDataAccess();
```
the "new TestDataAccess" should be replacing the "RedisAccessObject", not having both. Also, unless this is somehow invalidated by the way play does things, this should be set up in a way that allows us to give the controller, or if necessary an intermediary, the type of AbstractDataAccess to use, then we don't need to physically change the class between testing and the real thing. 

Might want to check if there is a "proper" way to do that in play but typical options are a setter method or a configuration file (so you can have a running config and a testing config)

reoliveira commented on app/services/dataAccess/TestDataAccess.java:
```diff
@@ -0,0 +1,44 @@
+ * Created by maria on 04/11/16.
+ */
+
+public class TestDataAccess extends AbstractDataAccess {
```
Is making actual sample post objects in here to send out part of a different pull request?

reoliveira commented on public/javascripts/hello.js:
```diff
@@ -2,23 +2,33 @@ if (window.console) {
   console.log("Welcome to your Play application's JavaScript!");
```
Super picky but the name of this file isn't really descriptive of what it does

reoliveira reviewed (changes requested):
Needs separation of testing state and live state (the data objects instantiated in the controllers) in order to be put on master.

MariaStrasky committed:
Creating a post object for my test class

edolinsky committed:
recompiled javascript protobuf libraries sans closure imports

reoliveira committed:
rebasing fe serialization to a new starting point

edolinsky committed:
added Json serializer, gson library

MariaStrasky committed:
Initial code for displaying posts on front end

MariaStrasky committed:
Have all post data being displayed for a fake post

MariaStrasky committed:
Removing some extra code that's no longer used

And added in a link to the posts's source so that when you click on the
post it brings you there

reoliveira committed:
cleanup for merge

reoliveira committed:
Merge branch 'master' into Passing-in-test-data-to-front-end
